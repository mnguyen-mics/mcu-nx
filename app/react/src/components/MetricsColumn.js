import React from 'react';
import PropTypes from 'prop-types';

function MetricsFocus(props) {

  const {
    metrics,
    isLoading,
  } = props;
  const height = 375;
  const nbOfVal = metrics ? metrics.length : 1;
  const cellHeight = height / nbOfVal;

  return (
    <div className="" style={{ paddingRight: '20px' }}>
      {metrics.map(metric => {
        return (
          <div key={metric.name} style={{ height: `${cellHeight}px` }}>
            <div style={{ fontSize: '0.9em', textAlign: 'left', color: '#919191' }}>{metric.name}</div>
            <div style={{ fontSize: '2em', fontWeight: 'bold', textAlign: 'left', color: 'rgba(0, 0, 0, 0.65)', lineHeight: '1.25' }}>{isLoading ? <i className="mcs-table-cell-loading" style={{ width: '130px' }} /> : metric.value}</div>
          </div>);
      })}
    </div>
  );
}

MetricsFocus.defaultProps = {
  isLoading: false,
};

MetricsFocus.propTypes = {
  metrics: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    value: PropTypes.string,
  }).isRequired).isRequired,
  isLoading: PropTypes.bool,
};

export default MetricsFocus;
