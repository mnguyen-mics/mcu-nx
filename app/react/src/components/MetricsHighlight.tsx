import * as React from 'react';
import { Col } from 'antd';

interface Metric {
  name?: string;
  value?: string;
}

interface MetricsColumnProps {
  metrics: Metric[];
  isLoading?: boolean;
}

class MetricsColumn extends React.Component<MetricsColumnProps> {
  static defaultProps = {
    isLoading: false,
  };

  render() {
    const { metrics, isLoading } = this.props;

    const nbOfVal: number = metrics ? metrics.length : 1;
    const cellSize: number = Math.round(24 / nbOfVal);

    return (
      <div className="mcs-metrics-column">
        {metrics.map(metric => {
          return (
            <Col key={metric.name} span={cellSize}>
              <div className="title">{metric.name}</div>
              <div className="metric">
                {isLoading ? (
                  <i
                    className="mcs-table-cell-loading"
                    style={{ width: '130px' }}
                  />
                ) : (
                  metric.value
                )}
              </div>
            </Col>
          );
        })}
      </div>
    );
  }
}

export default MetricsColumn;
