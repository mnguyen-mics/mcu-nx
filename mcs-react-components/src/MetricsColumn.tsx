import * as React from 'react';

export interface MetricsColumnProps {
  metrics: [{
    name?: string;
    value?: string;
  }];
  isLoading?: boolean;
}

const MetricsColumn: React.SFC<MetricsColumnProps> = props => {

  const {
    metrics,
    isLoading,
  } = props;

  const height = 375;
  const nbOfVal: number = metrics ? metrics.length : 1;
  const cellHeight: number = height / nbOfVal;

  return (
    <div className="p-r-20 mcs-metrics-column">
      {metrics.map(metric => {
        return (
          <div key={metric.name} style={{ height: `${cellHeight}px` }}>
            <div className="title">{metric.name}</div>
            <div className="metric">{isLoading ? <i className="mcs-table-cell-loading" style={{ width: '130px' }} /> : metric.value}</div>
          </div>);
      })}
    </div>
  );
};

MetricsColumn.defaultProps = {
  isLoading: false,
};

export default MetricsColumn;
