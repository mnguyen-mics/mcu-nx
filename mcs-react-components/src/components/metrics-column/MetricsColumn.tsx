import * as React from 'react';

export interface Metric {
  name: string;
  value?: string;
}

export interface MetricsColumnProps {
  metrics: Metric[];
  isLoading?: boolean;
}

class MetricsColumn extends React.Component<MetricsColumnProps> {
  static defaultProps = {
    isLoading: false,
  };

  render() {
    const { metrics, isLoading } = this.props;

    const height = 375;
    const nbOfVal: number = metrics ? metrics.length : 1;
    const cellHeight: number = height / nbOfVal;

    return (
      <div className="p-r-20 mcs-metrics-column">
        {metrics.map(metric => {
          return (
            <div key={metric.name} style={{ height: `${cellHeight}px` }}>
              <div className="title">{metric.name}</div>
              <div className="metric">
                {isLoading ? (
                  <i
                    className="mcs-table-cell-loading"
                    style={{ width: '130px' }}
                  />
                ) : (
                  metric.value
                )}
              </div>
            </div>
          );
        })}
      </div>
    );
  }
}

export default MetricsColumn;
