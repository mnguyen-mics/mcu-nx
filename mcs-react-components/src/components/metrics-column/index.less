.p-r-20 {
  padding-right: 20px;
}

.mcs-metrics-column {
  .title {
    font-size: 0.9em;
    text-align: left;
    color: fade(@heading-color, 40%);
  }
  .metric {
    font-size: 2em;
    font-weight: bold;
    text-align: left;
    color: fade(@heading-color, 85%);
    line-height: 1.25;
  }
}

.mcs-table-cell-loading {
    display: inline-block;
    font-style: normal;
    vertical-align: baseline;
    text-align: center;
    text-transform: none;
    line-height: 1;
    width: 100%;
    max-width: 100px;
    height: 8px;
    border-radius: 10px;
    background: linear-gradient(149deg, #dee4e8, #eceaea);
    background-size: 400% 400%;
    -webkit-animation: loadingAnimation 1s ease infinite;
    -moz-animation: loadingAnimation 1s ease infinite;
    -o-animation: loadingAnimation 1s ease infinite;
    animation: loadingAnimation 1s ease infinite;
  }