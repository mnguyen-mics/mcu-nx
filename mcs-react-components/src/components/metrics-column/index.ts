import MetricsColumn from './MetricsColumn';

export { MetricsColumnProps, Metric } from './MetricsColumn'

export default MetricsColumn;
