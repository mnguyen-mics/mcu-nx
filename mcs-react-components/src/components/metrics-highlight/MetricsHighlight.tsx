import * as React from 'react';
import { Col, Row } from 'antd';
import { MetricsColumnProps } from '../metrics-column';

class MetricsHighlight extends React.Component<MetricsColumnProps> {
  static defaultProps = {
    isLoading: false,
  };

  render() {
    const { metrics, isLoading } = this.props;

    const nbOfVal: number = metrics ? metrics.length : 1;
    const cellSize: number = Math.round(24 / nbOfVal);
    const prefixCls = 'mcs-metrics-column';

    return (
      <div className={prefixCls}>
        <Row>
          {metrics.map(metric => {
            return (
              <Col key={metric.name} span={cellSize}>
                <div className={`${prefixCls}_title`}>{metric.name}</div>
                <div className={`${prefixCls}_metric`}>
                  {isLoading ? (
                    <i
                      className={`${prefixCls}_loading`}
                      style={{ width: '130px' }}
                    />
                  ) : (
                    metric.value
                  )}
                </div>
              </Col>
            );
          })}
        </Row>
      </div>
    );
  }
}

export default MetricsHighlight;
