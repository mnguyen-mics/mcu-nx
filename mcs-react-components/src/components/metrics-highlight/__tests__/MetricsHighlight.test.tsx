import * as React from 'react';
import { MemoryRouter } from 'react-router';
import { IntlProvider } from 'react-intl';
import { render } from 'enzyme';
import { MetricsColumnProps } from '../../metrics-column/MetricsColumn';
import MetricsHighlight from '../MetricsHighlight';

it('renders the metrics highlight', () => {
  const props: MetricsColumnProps = {
    metrics: [
      { name: 'TestMetric1', value: '1' },
      { name: 'TestMetric2', value: '2' },
      { name: 'TestMetric3', value: '3' },
    ],
  };

  const component = render(
    <IntlProvider locale="en">
      <MemoryRouter>
        <MetricsHighlight {...props} />
      </MemoryRouter>
    </IntlProvider>,
  );
  expect(component).toMatchSnapshot();
});
