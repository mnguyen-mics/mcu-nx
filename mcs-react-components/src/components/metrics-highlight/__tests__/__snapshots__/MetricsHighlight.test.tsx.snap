// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the metrics highlight 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "mcs-metrics-column",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "ant-row",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "ant-col ant-col-8",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "mcs-metrics-column_title",
                },
                "children": Array [
                  Node {
                    "data": "TestMetric1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_metric",
                  },
                  "children": Array [
                    Node {
                      "data": "1",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Node {
                "attribs": Object {
                  "class": "mcs-metrics-column_metric",
                },
                "children": Array [
                  Node {
                    "data": "1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_title",
                  },
                  "children": Array [
                    Node {
                      "data": "TestMetric1",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "ant-col ant-col-8",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_title",
                  },
                  "children": Array [
                    Node {
                      "data": "TestMetric2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Node {
                    "attribs": Object {
                      "class": "mcs-metrics-column_metric",
                    },
                    "children": Array [
                      Node {
                        "data": "2",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_metric",
                  },
                  "children": Array [
                    Node {
                      "data": "2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Node {
                    "attribs": Object {
                      "class": "mcs-metrics-column_title",
                    },
                    "children": Array [
                      Node {
                        "data": "TestMetric2",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {
                  "class": "ant-col ant-col-8",
                },
                "children": Array [
                  Node {
                    "attribs": Object {
                      "class": "mcs-metrics-column_title",
                    },
                    "children": Array [
                      Node {
                        "data": "TestMetric3",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Node {
                      "attribs": Object {
                        "class": "mcs-metrics-column_metric",
                      },
                      "children": Array [
                        Node {
                          "data": "3",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  Node {
                    "attribs": Object {
                      "class": "mcs-metrics-column_metric",
                    },
                    "children": Array [
                      Node {
                        "data": "3",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": Node {
                      "attribs": Object {
                        "class": "mcs-metrics-column_title",
                      },
                      "children": Array [
                        Node {
                          "data": "TestMetric3",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Node {
            "attribs": Object {
              "class": "ant-col ant-col-8",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "mcs-metrics-column_title",
                },
                "children": Array [
                  Node {
                    "data": "TestMetric2",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_metric",
                  },
                  "children": Array [
                    Node {
                      "data": "2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Node {
                "attribs": Object {
                  "class": "mcs-metrics-column_metric",
                },
                "children": Array [
                  Node {
                    "data": "2",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_title",
                  },
                  "children": Array [
                    Node {
                      "data": "TestMetric2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "ant-col ant-col-8",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_title",
                  },
                  "children": Array [
                    Node {
                      "data": "TestMetric3",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Node {
                    "attribs": Object {
                      "class": "mcs-metrics-column_metric",
                    },
                    "children": Array [
                      Node {
                        "data": "3",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_metric",
                  },
                  "children": Array [
                    Node {
                      "data": "3",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Node {
                    "attribs": Object {
                      "class": "mcs-metrics-column_title",
                    },
                    "children": Array [
                      Node {
                        "data": "TestMetric3",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "ant-col ant-col-8",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_title",
                  },
                  "children": Array [
                    Node {
                      "data": "TestMetric1",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Node {
                    "attribs": Object {
                      "class": "mcs-metrics-column_metric",
                    },
                    "children": Array [
                      Node {
                        "data": "1",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_metric",
                  },
                  "children": Array [
                    Node {
                      "data": "1",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Node {
                    "attribs": Object {
                      "class": "mcs-metrics-column_title",
                    },
                    "children": Array [
                      Node {
                        "data": "TestMetric1",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          Node {
            "attribs": Object {
              "class": "ant-col ant-col-8",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "class": "mcs-metrics-column_title",
                },
                "children": Array [
                  Node {
                    "data": "TestMetric3",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_metric",
                  },
                  "children": Array [
                    Node {
                      "data": "3",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Node {
                "attribs": Object {
                  "class": "mcs-metrics-column_metric",
                },
                "children": Array [
                  Node {
                    "data": "3",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_title",
                  },
                  "children": Array [
                    Node {
                      "data": "TestMetric3",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "class": "ant-col ant-col-8",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_title",
                  },
                  "children": Array [
                    Node {
                      "data": "TestMetric2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Node {
                    "attribs": Object {
                      "class": "mcs-metrics-column_metric",
                    },
                    "children": Array [
                      Node {
                        "data": "2",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                Node {
                  "attribs": Object {
                    "class": "mcs-metrics-column_metric",
                  },
                  "children": Array [
                    Node {
                      "data": "2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Node {
                    "attribs": Object {
                      "class": "mcs-metrics-column_title",
                    },
                    "children": Array [
                      Node {
                        "data": "TestMetric2",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {
                  "class": "ant-col ant-col-8",
                },
                "children": Array [
                  Node {
                    "attribs": Object {
                      "class": "mcs-metrics-column_title",
                    },
                    "children": Array [
                      Node {
                        "data": "TestMetric1",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Node {
                      "attribs": Object {
                        "class": "mcs-metrics-column_metric",
                      },
                      "children": Array [
                        Node {
                          "data": "1",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  Node {
                    "attribs": Object {
                      "class": "mcs-metrics-column_metric",
                    },
                    "children": Array [
                      Node {
                        "data": "1",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": Node {
                      "attribs": Object {
                        "class": "mcs-metrics-column_title",
                      },
                      "children": Array [
                        Node {
                          "data": "TestMetric1",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
